storagenode/storagenodedb/storageusage.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storagenodedb
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storagenode/storageusage"
+)
+
+// StorageUsage returns storageusage.DB
+func (db *InfoDB) StorageUsage() storageusage.DB { return &storageusageDB{db} }
+
+// StorageUsage returns storageusage.DB
+func (db *DB) StorageUsage() storageusage.DB { return db.info.StorageUsage() }
+
+// storageusageDB storage usage DB
+type storageusageDB struct {
+	*InfoDB
+}
+
+// Store stores storage usage stamps to db replacing conflicting entries
+func (db *storageusageDB) Store(ctx context.Context, stamps []storageusage.Stamp) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	if len(stamps) == 0 {
+		return nil
+	}
+
+	query := `INSERT OR REPLACE INTO storage_usage(satellite_id, at_rest_total, timestamp) 
```

Comment:
Same here, i dont think there should at any time be an replace/update happening..

Reply:
I am not sure here, I can't not be sure that third party satellite has the same policy and intervals for rollups. We have no protocolo enforcement for that. And that's storagenodes side, where it can interact with basically any satellite on the network